### PR TITLE
Provide Interface to CDF Format Guide as Code

### LINKED ISSUE #62: Provide an interface to the CDF Format Guide
The HERMES yaml CDF Schema files closely encode the HERMES CDF Format Guide. By adding descriptions and possible examples to the schema yaml files we can provide better information when users are interested in learning more about the requirements and examples of specific attributes.

### PR DESCRIPTION
**Pull request by Alrobbertz**
closes #62

### CODE REVIEW

**Comment by Alrobbertz** [comment]
@ehsteve Looks like this one has the new builds

**Comment by Alrobbertz** [comment]
> @ehsteve Looks like this one has the new builds

It looks like the readthedocs build is failing due to missing C libraries :)

**Comment by ehsteve** [comment]
The doc build broke once we added spacepy which required the CDF libraries. I’ll look into fixing this.

**Comment by ehsteve** [comment]
For reviewers, you'll need to build documentation locally since our readthedocs builds are currently broken. See [our documentation](https://hermes-core.readthedocs.io/en/latest/dev-guide/docs.html#usage-and-building-the-documentation).

**Comment by ehsteve** [comment]
@Alrobbertz could update this from main to fix the rtd doc build please?

**Comment by ehsteve** [comment]
@tonyastrofig good question! So we already depend on astropy so using that functionality does not "cost us" anything. Astropy tables are like light pandas dataframes. They allow you to easily display and manipulate tabular data. Astropy uses them when displaying fits metadata (see this [learn topic](https://learn.astropy.org/tutorials/FITS-tables.html)) which is not so different than what we are doing. They also play well with jupyter notebooks and other formats like they can export to html or latex.

File: docs/user-guide/cdf_format_guide.rst [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+* default: (`str`) The default value used if none is provided
+* derived: (`bool`) Whether the attibute can be derived by the HERMES 
+  :py:class:`~hermes_core.util.schema.CDFSchema` class
+* required: (`bool`) Whether the attribute is required by HERMES standards
+* validate: (`bool`) Whether the attribute is included in the 
+  :py:func:`~hermes_core.util.validation.validate` checks (Note, not all attributes that 
+  are required are validated)
+* overwrite: (`bool`) Whether the :py:class:`~hermes_core.util.schema.CDFSchema`
+  attribute derivations will overwrite an existing attribute value with an updated 
+  attribute value from the derivation process.
```
Comment (ehsteve) [inline docs/user-guide/cdf_format_guide.rst:210]:
```suggestion
  attribute value from the derivation process. Note that this table is derived from :file:`hermes_core/data/hermes_default_global_cdf_attrs_schema.yaml`
```

**Review by ehsteve — APPROVED** [review]

**Pull request merged**